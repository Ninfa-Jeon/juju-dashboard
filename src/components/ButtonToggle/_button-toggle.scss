@import "vanilla-framework/scss/vanilla";

.p-button-toggle {
  align-items: center;
  column-gap: 1rem;
  display: grid;
  font-size: 0.875rem;
  grid-template-columns: 0.25fr 0.75fr;
  padding: 0.25rem;

  &__label {
    display: inline-block;
    text-align: right;
  }

  &__buttons {
    border: 1px solid $color-mid-light;
    border-radius: 4px;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }

  &__button {
    border: none;
    border-right: 1px solid $color-mid-light;
    font-size: 0.875rem;
    margin: 0 !important; // Required to override Vanilla button:not(:last-of-type):not(:only-of-type)
    padding: 0 0.25rem;
    text-align: center;

    &.is-selected {
      background: $color-mid-x-light;
    }

    &:last-child {
      border: 0;
    }
  }
}
