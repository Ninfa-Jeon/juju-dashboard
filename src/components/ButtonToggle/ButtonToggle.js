import React from "react";

import "./_button-toggle.scss";

const ButtonToggle = () => {
  return (
    <div className="p-button-toggle">
      <span className="p-button-toggle__label">Group by:</span>
      <div className="p-button-toggle__buttons">
        <button className="p-button-toggle__button is-selected">Status</button>
        <button className="p-button-toggle__button">Cloud</button>
        <button className="p-button-toggle__button">Owner</button>
      </div>
    </div>
  );
};

export default ButtonToggle;
