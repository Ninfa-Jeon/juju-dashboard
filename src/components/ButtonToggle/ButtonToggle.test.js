import React from "react";
import { shallow } from "enzyme";

import ButtonToggle from "./ButtonToggle";

describe("Button toggle", () => {
  it("renders without crashing and matches snapshot", () => {
    const wrapper = shallow(<ButtonToggle />);
    expect(wrapper).toMatchSnapshot();
  });
});
