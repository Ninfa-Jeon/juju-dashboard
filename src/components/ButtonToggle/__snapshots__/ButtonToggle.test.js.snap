// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button toggle renders without crashing and matches snapshot 1`] = `
<div
  className="p-button-toggle"
>
  <div
    className="p-button-toggle__inner"
  >
    <span
      className="p-button-toggle__label"
    >
      Group by:
    </span>
    <div
      className="p-button-toggle__buttons"
    >
      <button
        className="p-button-toggle__button is-selected"
      >
        Status
      </button>
      <button
        className="p-button-toggle__button"
      >
        Cloud
      </button>
      <button
        className="p-button-toggle__button"
      >
        Owner
      </button>
    </div>
  </div>
</div>
`;
